File: fsharp/difference-of-squares/DifferenceOfSquares.fs
module DifferenceOfSquares

    let private square (n:int) = n * n

    let squareOfSum (number:int):int =
        seq{1..number}
        |> Seq.sum
        |> square

    let sumOfSquares (number:int):int =
        seq{1..number}
        |> Seq.sumBy square

    let differenceOfSquares (number:int):int = squareOfSum number - sumOfSquares number

File: solutions/scala/difference-of-squares/1/src/main/scala/DifferenceOfSquares.scala
object DifferenceOfSquares
{

  private def square (n:Int):Int = n * n

  def sumOfSquares(n: Int): Int = n * (n + 1) * (2 * n + 1) / 6

  def squareOfSum(n: Int): Int = square(Seq.range(1, n + 1).sum())

  def differenceOfSquares(n: Int): Int =  squareOfSum(n) - sumOfSquares(n)
}
